### ISSUE TITLE
Issue related to add value residual

### ISSUE THREAD

**Issue by pcunwa** [lucidrains/BS-RoFormer#39]
Thanks for the interesting update. However, there seems to be a problem: the weight and bias of the to_value_residual_mix are not updated. Perhaps the gradient is not flowing properly. I fixed Attention module as follows and it is now resolved.
However, it is not clear whether this modification is appropriate.
```
class Attention(Module):
    def __init__(
        self,
        dim,
        heads = 8,
        dim_head = 64,
        dropout = 0.,
        rotary_embed = None,
        flash = True,
        learned_value_residual_mix = False
    ):
        super().__init__()
        self.heads = heads
        self.scale = dim_head **-0.5
        dim_inner = heads * dim_head

        self.rotary_embed = rotary_embed
        self.attend = Attend(flash = flash, dropout = dropout)
        self.norm = RMSNorm(dim)
        self.to_qkv = nn.Linear(dim, dim_inner * 3, bias = False)
        
        self.has_value_residual = learned_value_residual_mix
        self.to_value_residual_mix = nn.Linear(dim, heads) if learned_value_residual_mix else None
        self.to_gates = nn.Linear(dim, heads)
        self.to_out = nn.Sequential(
            nn.Linear(dim_inner, dim, bias = False),
            nn.Dropout(dropout)
        )

    def forward(self, x, value_residual = None):
        x = self.norm(x)
        q, k, v = rearrange(self.to_qkv(x), 'b n (qkv h d) -> qkv b h n d', qkv = 3, h = self.heads)

        current_v = v.clone()

        if exists(value_residual) and self.has_value_residual:
            mix = self.to_value_residual_mix(x)
            mix = rearrange(mix, 'b n h -> b h n 1').sigmoid()
            v = v.lerp(value_residual, mix)

        if exists(self.rotary_embed):
            q = self.rotary_embed.rotate_queries_or_keys(q)
            k = self.rotary_embed.rotate_queries_or_keys(k)

        out = self.attend(q, k, v)
        gates = self.to_gates(x)
        out = out * rearrange(gates, 'b n h -> b h n 1').sigmoid()
        out = rearrange(out, 'b h n d -> b n (h d)')
        
        return self.to_out(out), current_v
```

**Comment by lucidrains** [comment]
@pcunwa oops! thank you

could you see if the latest commit lines up with yours?

**Comment by lucidrains** [comment]
@pcunwa no problem, you should see a nice improvement from this added research finding.
